### PR TITLE
Adding missing_items report query

### LINKED ISSUE #422: RA/UM - build Missing Items report query
https://issues.folio.org/browse/UXPROD-2853

### PR DESCRIPTION
**Pull request by amzoss**
Runs on `folio_snapshot_20210525_missing` database. Fixes #422

### CODE REVIEW

**Review by adoerrer — APPROVED** [review]
Looks good to go, thanks!

**Review by jenmawe — APPROVED** [review]

File: sql/report_queries/missing_items/missing_items.sql [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        loan_return_date 
+    FROM
+        ranked_loans
+    WHERE
+        ranked_loans.return_date_ranked = 1
+),
+item_notes_list AS (
+    SELECT
+        itn.item_id,
+        string_agg(DISTINCT itn.note, '|'::text) AS notes_list
```
Comment (nassibnassar) [inline sql/report_queries/missing_items/missing_items.sql:60]: ::text is not needed for inline string literals

File: sql/report_queries/missing_items/missing_items.sql [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        ie.instance_id 
+    FROM 
+        item_subset AS its
+        LEFT JOIN folio_reporting.holdings_ext AS he ON its.holdings_record_id = he.holdings_id
+        LEFT JOIN folio_reporting.instance_ext AS ie ON he.instance_id = ie.instance_id
+),
+publication_dates_list AS (
+    SELECT
+        ip.instance_id,
+        string_agg(DISTINCT date_of_publication, '|'::text) AS publication_dates_list
```
Comment (nassibnassar) [inline sql/report_queries/missing_items/missing_items.sql:78]: ::text is not needed for inline string literals

File: sql/report_queries/missing_items/missing_items.sql [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    FROM
+        folio_reporting.instance_publication AS ip
+        RIGHT JOIN instance_subset AS ins ON ip.instance_id = ins.instance_id
+    GROUP BY
+        ip.instance_id
+)
+    ---------- MAIN QUERY ----------
+SELECT
+    (SELECT start_date::varchar FROM parameters) || 
+        ' to '::varchar ||
```
Comment (nassibnassar) [inline sql/report_queries/missing_items/missing_items.sql:88]: ::varchar is not needed for inline string literals

**Review by nassibnassar — CHANGES_REQUESTED** [review]

File: sql/report_queries/missing_items/missing_items.sql (hunk 1 above)
Comment (amzoss) [reply sql/report_queries/missing_items/missing_items.sql]: removed in latest commit

File: sql/report_queries/missing_items/missing_items.sql (hunk 3 above)
Comment (amzoss) [reply sql/report_queries/missing_items/missing_items.sql]: removed in latest commit

File: sql/report_queries/missing_items/missing_items.sql (hunk 2 above)
Comment (amzoss) [reply sql/report_queries/missing_items/missing_items.sql]: removed in latest commit

**Review by nassibnassar — APPROVED** [review]

**Pull request merged**